Lambda-School-Labs/CS10-quizzer
# VALIDATE TEACHERS TRYING TO CREATE A NEW CLASS

# Description

Validates any Teacher(user) trying to create a new Class.
This does not contain REAL error handling but will stop non-authenticated or users who do NOT exist from creating a new Class for now.

## Type of change

# How Has This Been Tested?

- `cd server`
- `pipenv shell`
- `./manage.py migrate`
- ./manage.py runserver`
- Navigate to `http://127.0.0.1:8000/graphiql/`
- Create Teacher

```
mutation {
  createTeacher(ClassID:"1", TeacherName:"Mr. Benefield") {
    teacher {
      TeacherName
      ClassID
    }
    jwtString
  }
}
```

- Copy the JWT into clipboard (this simulates saving JWT on localStorage)
- Create new class

```
mutation {
  createClass(ClassName:"New Class", encJwt:"<< JWT >>") {
    newClass {
      ClassName
    }
  }
}
```

- Check class was created

```
{
  classes {
    ClassesID
    ClassName
    createdAt
    lastModified
  }
}
```

# Checklist:
